frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx

```diff
@@ -0,0 +1,301 @@
+  const {
+    network,
+    nads,
+    usedInterfacesNames,
+    usedMultusNetworkNames,
+    allowPodNetwork,
+    onSubmit,
+    inProgress,
+    errorMessage,
+    handlePromise,
+    close,
+    cancel,
+  } = props;
+  const asId = prefixedID.bind(null, 'nic');
+  const nic = props.nic || NetworkInterfaceWrapper.EMPTY;
+  const isEditing = nic !== NetworkInterfaceWrapper.EMPTY;
+
+  const [name, setName] = React.useState<string>(
+    nic.getName() || getSequenceName('nic', usedInterfacesNames),
+  );
+  const [model, setModel] = React.useState<NetworkInterfaceModel>(
+    nic.getModel() || (isEditing ? null : NetworkInterfaceModel.VIRTIO),
+  );
+  const [resultNetwork, setResultNetwork] = React.useState<NetworkWrapper>(network);
+  const [interfaceType, setInterfaceType] = React.useState<NetworkInterfaceType>(nic.getType());
+  const [macAddress, setMacAddress] = React.useState<string>(nic.getMACAddress() || '');
+
+  const resultNIC = NetworkInterfaceWrapper.initializeFromSimpleData({
+    name,
+    model,
+    interfaceType,
+    macAddress,
+  });
+
+  const {
+    validations: { name: nameValidation, macAddress: macAddressValidation },
+    isValid,
+    hasAllRequiredFilled,
+  } = validateNIC(resultNIC, resultNetwork, { usedInterfacesNames });
+
+  const [showUIError, setShowUIError] = useShowErrorToggler(false, isValid, isValid);
+
+  const onNetworkChoiceChange = (newNetworkChoice: NetworkWrapper) => {
+    if (newNetworkChoice.isPodNetwork()) {
+      setMacAddress('');
+    }
+
+    if (!interfaceType || !newNetworkChoice.getType().allowsInterfaceType(interfaceType)) {
+      setInterfaceType(newNetworkChoice.getType().getDefaultInterfaceType());
+    }
+    setResultNetwork(newNetworkChoice);
+  };
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    if (isValid) {
+      // eslint-disable-next-line promise/catch-or-return
+      handlePromise(
+        onSubmit(
+          resultNIC,
+          NetworkWrapper.mergeWrappers(
+            resultNetwork,
+            NetworkWrapper.initializeFromSimpleData({ name: resultNIC.getName() }),
+          ),
+        ),
+      ).then(close);
+    } else {
+      setShowUIError(true);
+    }
+  };
+
+  return (
+    <div className="modal-content">
+      <ModalTitle>{isEditing ? 'Edit' : 'Add'} Network Interface</ModalTitle>
+      <ModalBody>
+        <Form>
+          <FormRow
+            title="Name"
+            fieldId={asId('name')}
+            isRequired
+            isLoading={!usedInterfacesNames}
+            validationMessage={nameValidation && nameValidation.message}
+            validationType={nameValidation && nameValidation.type}
+          >
+            <TextInput
+              isValid={!isValidationError(nameValidation)}
+              isDisabled={!usedInterfacesNames || inProgress}
+              isRequired
+              id={asId('name')}
+              value={name}
+              onChange={(v) => setName(v)}
```

Comment:
nit:
```suggestion
              onChange={setName}
```

Reply:
this does not work because the second argument is event and React.useState callback complains about that